tensorflow_addons/seq2seq/decoder.py

```diff
@@ -441,8 +441,9 @@ def _maybe_copy_state(new, cur):
                 else:
                     new.set_shape(cur.shape)
                     pass_through = (new.shape.ndims == 0)
-                broadcast_finished = tf.broadcast_to(
-                    tf.expand_dims(finished, axis=-1), new.shape)
+                if not pass_through:
+                    broadcast_finished = tf.broadcast_to(
+                        tf.expand_dims(finished, axis=-1), new.shape)
                 return new if pass_through else tf.where(
```

Comment:
Let's merge this `if` statement with the one above, e.g:

```python
if not pass_through:
    ...
    return tf.where(...)
else:
    return new
```

Reply:
done.